lib/nbp/push_connector.rb

```diff
@@ -0,0 +1,153 @@
+# frozen_string_literal: true
+
+require 'singleton'
+require 'concurrent'
+
+module Nbp
+  class PushConnector
+    include Singleton
+
+    class Error < StandardError; end
+    class SettingsError < Error; end
+    class ServerError < Error; end
+
+    def initialize
+      super
+      @token = Concurrent::ThreadLocalVar.new
+      @token_expiration = Concurrent::ThreadLocalVar.new
+
+      create_source! unless source_exists?
+    end
+
+    def self.enabled?
+      @enabled ||= Settings.nbp&.push_connector&.enable || false
```

Comment:
Memoizing a boolean value is difficult, and we should probably switch to [another method](https://dev.to/codeandclay/a-memoization-gotcha-14c6).

Reply:
I pushed a fix :+1: